File: TRUVD/geometric.m
function X=geometric(m,n,kappa)
     p=min([m,n]);
     lnkappa=-log(kappa)/(p-1);
     exponents=(0:p-1)*lnkappa;
     sigma=exp(exponents);
     if m > n
        U=orth(randn(m,n));  V=orth(randn(n,n));
    else
	U=orth(randn(m,m));  V=orth(randn(n,m));
    end;
    X=U*diag(sigma)*V';

File: TURVDinPython/geometric.py
import scipy.linalg as lin
import numpy as np
import math
import sys

np.set_printoptions(threshold=sys.maxsize)

def geometry(m,n,kappa):
    p = min([m,n])
    Inkappa = -math.log(kappa)/(p-1)
    exponents = list(range(0,p))
    sigma = [math.exp(i * Inkappa) for i in exponents]

    if m > n:
        U = lin.orth(np.random.rand(m,n))
        V = lin.orth(np.random.rand(n,n))
    else:
        U = lin.orth(np.random.rand(m,m))
        V = lin.orth(np.random.rand(n,m))
    # sys.stdout = open('output1.txt', 'w')
    X = np.dot(np.dot(U, np.diag(sigma)), V.transpose())
    print(X)
    return X



